internal/command/org_domain.go

```diff
@@ -390,3 +391,56 @@ func (c *Commands) getOrgDomainWriteModel(ctx context.Context, orgID, domain str
 	}
 	return domainWriteModel, nil
 }
+
+type OrgDomainVerified struct {
+	OrgID    string
+	Domain   string
+	Verified bool
+}
+
+func (c *Commands) searchOrgDomainVerifiedByDomain(ctx context.Context, domain string) (_ *OrgDomainVerified, err error) {
+	ctx, span := tracing.NewSpan(ctx)
+	defer func() { span.EndWithError(err) }()
+
+	conditions := []map[eventstore.FieldType]any{
+		{
+			eventstore.FieldTypeAggregateType: org.AggregateType,
+			eventstore.FieldTypeObjectType:    org.OrgDomainSearchType,
+			eventstore.FieldTypeFieldName:     org.OrgDomainDomainSearchField,
+			eventstore.FieldTypeValue:         domain,
+		},
+		{
+			eventstore.FieldTypeAggregateType: org.AggregateType,
+			eventstore.FieldTypeObjectType:    org.OrgDomainSearchType,
+			eventstore.FieldTypeFieldName:     org.OrgDomainVerifiedSearchField,
+		},
+	}
+
+	results, err := c.eventstore.Search(ctx, conditions...)
+	if err != nil {
+		return nil, err
+	}
+	if len(results) == 0 {
+		_ = projection.OrgDomainVerifiedFields.Trigger(ctx)
+		results, err = c.eventstore.Search(ctx, conditions...)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
I'm not sure if a Trigger here is desired. The code calling this has the pupose of checking that a user suffix does not exist on another org. That means non-exsitence is actualy the "normal" flow. So as a consequence we might be triggering here with no added benefit.

Isn't it possible to do time based triggers, like with the projections?